src/config_objects.py

```diff
@@ -3,280 +3,104 @@
 
 class datasetConfig:
     
-    def __init__(self, id: str = "", type: DatasetType = "", title: str = "", description: str = "", topics: list[str] = [],
-                 next_release: str = "", keywords: list[str] = [], QMI_href: str = "", contact_name: str = "", contact_email: str = "",
-                 contact_telephone: str = "", publisher_name: str = "", publisher_href: str = ""):
-        self._id: str = id
-        self._type: DatasetType = DatasetType(type)
-        self._title: str = title
-        self._description: str = description
-        
-        self._topics: list[str] = topics
-        self._license: str = "Open Government License v3.0"
-        
-        self._next_release: str = next_release # This field is designated a string on the plan but both date related fields in edition are datetime objects. I believe this should also be a datetime object, both for the sake of consistancy and to make error checking easier.
-        self._keywords: list[str] = keywords
-        
-        self._QMI: dict = {"href": QMI_href}
-        self._Contact: dict = {"name": contact_name, 
-                              "email": contact_email,
-                              "telephone": contact_telephone}
-        self._Publisher: dict = {"name": publisher_name,
-                                "href": publisher_href}
-        
-        def get_id(self) -> str:
-            return self._id
-        
-        def set_id(self, new_id: str):
-            self._id = new_id
+    def __init__(self):
+
+        self._dataset_metadata = {
+            "id": "",
+            "type": DatasetType(),
+            "title": "",
+            "description": "",
+            "topics": [],
+            "license": "Open Government License v3.0",
+            "next_release": "",
+            "keywords": [],
+            "QMI": {"href": ""},
+            "contact": {"name": "", "email": "", "telephone": ""},
+            "publisher": {"name": "", "href": ""}
+        }
             
-        def get_type(self) -> DatasetType:
-            return self._type
-        
-        def set_type(self, new_type: str): 
+    def import_from_dict(self, new_meta_data: dict):
```

Comment:
This logic needs to change - some keys will be optional, and people might want to populate part of the config this way and then add a couple of fields separately. The condition should be `in` rather than `==`